### PR TITLE
[CALCITE-3482] Equality of nested ROWs returns false for identical literal value

### PR DESCRIPTION
**Pull request by rubenada**
Jira ticket: [CALCITE-3482](https://issues.apache.org/jira/browse/CALCITE-3482)

### CODE REVIEW

File: core/src/test/resources/sql/struct.iq [hunk 1]
```diff
@@ -47,4 +47,40 @@ select distinct * from (values
...
+!ok
+
+# [CALCITE-3482] Equality of nested ROWs returns false for identical literal value
+select * from
+(values
+    (1, ROW(2, ROW(4,2))),
+    (2, ROW(2, ROW(4,4))),
+    (3, ROW(2, ROW(1,4))),
+    (4, ROW(2, ROW(4,4)))) as t(id,struct)
+where t.struct = ROW(2, ROW(4,4));
```
Comment (hsyuan) [inline core/src/test/resources/sql/struct.iq]: Can we give them different values, e.g. (3,4) and (4,3)? In order to verify that the ordering does matter.

**Comment by danny0405** [comment]
Nice job, @rubenada !

File: core/src/test/resources/sql/struct.iq (hunk 1 above)
Comment (rubenada) [reply core/src/test/resources/sql/struct.iq]: Sure, done.

**Comment by rubenada** [comment]
Thanks for your review @danny0405  @hsyuan !

**Pull request merged**